crates/services/p2p/src/discovery.rs

```diff
@@ -238,27 +177,26 @@ impl NetworkBehaviour for DiscoveryBehaviour {
                             self.kademlia.add_address(&peer_id, multiaddr);
```

Comment:
Hmm, if `list.len() + self.connected_peers.len() > self.max_peers_connected` we will not handle this case=) After looking at how `add_address` works, I think it is okay to call `add_address` without any limitation(I think we can remoev `if self.connected_peers.len() < self.max_peers_connected` check) because it only updates the routing table.

Reply:
Yeah. 

````
                    if self.connected_peers.len() < self.max_peers_connected {
                        for (peer_id, multiaddr) in list {
                            self.kademlia.add_address(&peer_id, multiaddr);
                        }
                    }
```
doesn't even work as intended, I think, because it only checks before you add the first peer.